shopfloor/services/zone_picking.py

```diff
@@ -1135,6 +1145,11 @@ def set_destination(
 
         # When the barcode is a package
         package = search.package_from_scan(barcode)
+        if package and not allow_package:
```

Comment:
```suggestion
        if package:
           if not self.work.menu.allow_alternative_destination_package and move_line.result_package_id and move_line.result_package_id != package:
```

Reply:
We could also add a check in case of partial that we are effectively putting on a different package
```
if accept_only_package and move_line.package_id == package:
  message = ...dest_package_not_valid(package)
```